modules/service/permissioned-data-service.js

```diff
@@ -242,6 +243,28 @@ class PermissionedDataService {
         });
         await Promise.all(promises);
     }
+
+    getPublicKeyData() {
```

Comment:
These features have nothing directly to do with permissioned data, so I'd recommend putting them somewhere else, like some NetworkUtilities or something network related, since permissioned data service should have more to do with the datasets and not how they're transferred